core/src/stages/compute_ik.cpp

```diff
@@ -395,15 +418,22 @@ void ComputeIK::compute() {
 			rviz_marker_tools::appendFrame(solution.markers(), target_pose_msg, 0.1, "ik frame");
 			rviz_marker_tools::appendFrame(solution.markers(), ik_pose_msg, 0.1, "ik frame");
 
-			if (succeeded && i + 1 == ik_solutions.size())
+			if (maximize_clearance) {
+				// compute cost as 1. / clearance (larger clearance is better)
+				double clearance = ik_solutions.back().second;
+				if (clearance > 0.)
+					solution.setCost(1. / (clearance + 1e-4));
+				else
+					solution.markAsFailure();
```

Comment:
This section should be moved into the `if (succeeded && i + 1 == ik_solutions.size())` branch,
which indicates that we have found a valid IK solution at all.